Clean-up process-css-pseudo-selectors.py

nt1m opened the pull request:
#### 3cd087d87fd5299b8c7f544ff2d377decfbc408e
<pre>
Clean-up process-css-pseudo-selectors.py
<a href="https://bugs.webkit.org/show_bug.cgi?id=267152">https://bugs.webkit.org/show_bug.cgi?id=267152</a>
<a href="https://rdar.apple.com/120560220">rdar://120560220</a>

Reviewed by Ryosuke Niwa.

- Fix copyright header indentation and remove new line before it.
- Use f-strings instead of &apos;&apos;.format() for readability
- Use dictionary.get() whenever possible when filling in default values for missing keys
- Make argument parsing more similar to process-css-properties.py and process-css-values.py
- Rename &quot;condition&quot; to &quot;conditional&quot; to match other files
- Rename &quot;ShadowPseudoElement&quot; to &quot;UserAgentPart&quot;

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/CSSPseudoSelectors.json:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/parser/MutableCSSSelector.cpp:
(WebCore::MutableCSSSelector::parsePseudoElementSelector):
* Source/WebCore/css/process-css-pseudo-selectors.py:
(InputValidator.check_field_documentation):
(InputValidator.validate_fields):
(write_copyright_header):

Canonical link: <a href="https://commits.webkit.org/272720@main">https://commits.webkit.org/272720@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/70062a2f82c08eb4e66738b0c34b433cbf14f874

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32822 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11571 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34684 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35433 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29645 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33748 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13914 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8754 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/29095 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33265 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9756 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/29309 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8495 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8637 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29266 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36764 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/28080 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29814 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29670 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34757 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8755 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6724 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32629 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10444 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9366 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4233 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9377 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/css/process-css-pseudo-selectors.py:
```diff
@@ -626,75 +614,68 @@ def write_is_pseudo_element_enabled(self, writer, values):
-        # Generate code for shadow pseudo elements.
-        def write_condition_cases_for_shadow_elements(map):
+        # Generate code for user agent parts.
+        def write_condition_cases_for_user_agent_parts(map):
```
Maybe write_*conditional*_cases_for_user_agent_parts?

nt1m replied:
Changed it to `write_cases_for_user_agent_parts`

rniwa reviewed (approved):
r=me assuming we get green EWSes.

nt1m committed:
Clean-up process-css-pseudo-selectors.py
https://bugs.webkit.org/show_bug.cgi?id=267152
rdar://120560220

Reviewed by Ryosuke Niwa.

- Fix copyright header indentation and remove new line before it.
- Use f-strings instead of ''.format() for readability
- Use dictionary.get() whenever possible when filling in default values for missing keys
- Make argument parsing more similar to process-css-properties.py and process-css-values.py
- Rename "condition" to "conditional" to match other files
- Rename "ShadowPseudoElement" to "UserAgentPart"

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/css/CSSPseudoSelectors.json:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/parser/MutableCSSSelector.cpp:
(WebCore::MutableCSSSelector::parsePseudoElementSelector):
* Source/WebCore/css/process-css-pseudo-selectors.py:
(InputValidator.check_field_documentation):
(InputValidator.validate_fields):
(write_copyright_header):

Canonical link: https://commits.webkit.org/272720@main
